File: cpp/347.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> hashmap;
        for (int num : nums) {
            auto pos = hashmap.find(num);
            if (pos == hashmap.end()) hashmap[num] = 1;
            pos->second += 1;
        }
        // 排序
        vector<pair<int, int>> vec(hashmap.begin(), hashmap.end());
        sort(vec.begin(), vec.end(), [](const pair<int, int> &a, const pair<int, int> &b) {
            return b.second < a.second;
        });
        vector<int> res;
        for (int i = 0; i < k; ++k) {
            res.push_back(vec[i].first);
        }
        return res;
    }
};

File: python/347.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        for v in nums:
            if v in hashmap:
                hashmap[v] += 1
            else:
                hashmap[v] = 1
        
        sorted_map = sorted(hashmap.items(), key=lambda x: x[1], reverse=True)

        res = []
        for i in range(k):
            res.append(sorted_map[i][0])
        return res